### PR TITLE
[DOC] additional overview content for MirrorMaker 2.0

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**

Additional MirrorMaker 2.0 content for the Overview guide.
More on how to configure the MirrorMaker 2.0 resource, plus an example.

NOTE: We can further de-emphasize the earlier version of MirrorMaker if required.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Update/write design documentation in `./design`
- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

File: documentation/modules/overview/con-configuration-points-mirrormaker.adoc [hunk 1]
```diff
@@ -33,29 +19,96 @@ MirrorMaker 2.0 uses:
...
 * Target cluster configuration to output data to the target cluster
 
-A MirrorMaker 2.0 `MirrorSourceConnector` custom resource replicates topics from a source cluster to a target cluster.
+[discrete]
+=== Cluster configuration
+
+You configure a `KafkaMirrorMaker2` resource to define the Kafka Connect deployment, including the connection details of the source and target clusters,
+and start running a set of MirrorMaker 2.0 connectors to make the connection.
+
+Topic configuration is automatically synchronized between source and target clusters according to the topics defined in the MirrorMaker2 custom resource. Configuration changes are propagated to remote topics so that new topics and partitions are detected and created.
```
Comment (ppatierno) [inline documentation/modules/overview/con-configuration-points-mirrormaker.adoc:28]: "... in the KafkaMirrorMaker2 custom resource ..."

File: documentation/modules/overview/con-configuration-points-mirrormaker.adoc [hunk 2]
```diff
@@ -7,21 +7,7 @@
...
-
-MirrorMaker uses:
-
-* Consumer configuration to consume data from the source cluster
-* Producer configuration to output data to the target cluster
-
-Consumer and producer configuration includes any authentication and encryption settings.
-
-A _whitelist_ defines the topics to mirror from a source to a target cluster.
+You can use {ProductName} with MirrorMaker 2.0, though the earlier version of MirrorMaker continues to be supported.
```
Comment (laidan6000) [inline documentation/modules/overview/con-configuration-points-mirrormaker.adoc:10]:
```suggestion
You can use {ProductName} with MirrorMaker 2.0, although the earlier version of MirrorMaker continues to be supported.
```

File: documentation/modules/overview/con-configuration-points-mirrormaker.adoc [hunk 3]
```diff
@@ -33,29 +19,97 @@ MirrorMaker 2.0 uses:
 * Source cluster configuration to consume data from the source cluster
 * Target cluster configuration to output data to the target cluster
 
-A MirrorMaker 2.0 `MirrorSourceConnector` custom resource replicates topics from a source cluster to a target cluster.
+[discrete]
+=== Cluster configuration
+
+You configure a `KafkaMirrorMaker2` resource to define the Kafka Connect deployment, including the connection details of the source and target clusters,
+and start running a set of MirrorMaker 2.0 connectors to make the connection.
```
Comment (laidan6000) [inline documentation/modules/overview/con-configuration-points-mirrormaker.adoc:26]:
```suggestion
and then run a set of MirrorMaker 2.0 connectors to make the connection.
```

File: documentation/modules/overview/con-configuration-points-mirrormaker.adoc [hunk 4]
```diff
@@ -33,29 +19,97 @@ MirrorMaker 2.0 uses:
...
+
+Topic configuration is automatically synchronized between source and target clusters according to the topics defined in the `KafkaMirrorMaker2` custom resource.
```
Comment (laidan6000) [inline documentation/modules/overview/con-configuration-points-mirrormaker.adoc:28]:
```suggestion
Topic configuration is automatically synchronized between the source and target clusters according to the topics defined in the `KafkaMirrorMaker2` custom resource.
```

File: documentation/modules/overview/con-configuration-points-mirrormaker.adoc [hunk 5]
```diff
@@ -33,29 +19,97 @@ MirrorMaker 2.0 uses:
 * Source cluster configuration to consume data from the source cluster
 * Target cluster configuration to output data to the target cluster
 
-A MirrorMaker 2.0 `MirrorSourceConnector` custom resource replicates topics from a source cluster to a target cluster.
+[discrete]
+=== Cluster configuration
+
+You configure a `KafkaMirrorMaker2` resource to define the Kafka Connect deployment, including the connection details of the source and target clusters,
```
Comment (laidan6000) [inline documentation/modules/overview/con-configuration-points-mirrormaker.adoc:25]:
```suggestion
You configure a `KafkaMirrorMaker2` custom resource to define the Kafka Connect deployment, including the connection details of the source and target clusters,
```

File: documentation/modules/overview/con-configuration-points-mirrormaker.adoc [hunk 6]
```diff
@@ -33,29 +19,97 @@ MirrorMaker 2.0 uses:
...
+and start running a set of MirrorMaker 2.0 connectors to make the connection.
+
+Topic configuration is automatically synchronized between source and target clusters according to the topics defined in the `KafkaMirrorMaker2` custom resource.
+Configuration changes are propagated to remote topics so that new topics and partitions are detected and created.
+Topic replication is defined using regular expression patterns to _whitelist_ or _blacklist_ topics.
+
+MirrorMaker 2.0 connectors and related internal topics help manage the transfer and synchronization of data between the clusters.
```
Comment (laidan6000) [inline documentation/modules/overview/con-configuration-points-mirrormaker.adoc:32]:
```suggestion
The following MirrorMaker 2.0 connectors and related internal topics help manage the transfer and synchronization of data between the clusters.
```

File: documentation/modules/overview/con-configuration-points-mirrormaker.adoc [hunk 7]
```diff
@@ -33,29 +19,97 @@ MirrorMaker 2.0 uses:
...
+
+MirrorSourceConnector:: _MirrorSourceConnector_ creates remote topics from the originating cluster.
```
Comment (laidan6000) [inline documentation/modules/overview/con-configuration-points-mirrormaker.adoc:34]:
```suggestion
MirrorSourceConnector:: A _MirrorSourceConnector_ creates remote topics from the source cluster.
```

File: documentation/modules/overview/con-configuration-points-mirrormaker.adoc [hunk 8]
```diff
@@ -33,29 +19,97 @@ MirrorMaker 2.0 uses:
...
+MirrorCheckpointConnector:: _MirrorCheckpointConnector_ tracks and maps offsets for specified consumer groups using an _offset sync_ topic and _checkpoint_ topic.
```
Comment (laidan6000) [inline documentation/modules/overview/con-configuration-points-mirrormaker.adoc:35]:
```suggestion
MirrorCheckpointConnector:: A _MirrorCheckpointConnector_ tracks and maps offsets for specified consumer groups using an _offset sync_ topic and _checkpoint_ topic.
```

File: documentation/modules/overview/con-configuration-points-mirrormaker.adoc [hunk 9]
```diff
@@ -33,29 +19,97 @@ MirrorMaker 2.0 uses:
...
+The offset sync topic maps the source and target offsets for replicated topic partitions from record metadata.
+A checkpoint is emitted from each source cluster and replicated in the target cluster through the checkpoint topic.
+The checkpoint topic maps the last committed offset in the source and target cluster for replicated topic partitions in each consumer group.
+MirrorHeartbeatConnector:: _MirrorHeartbeatConnector_ periodically checks connectivity between clusters.
```
Comment (laidan6000) [inline documentation/modules/overview/con-configuration-points-mirrormaker.adoc:39]:
```suggestion
MirrorHeartbeatConnector:: A _MirrorHeartbeatConnector_ periodically checks connectivity between clusters.
```

**Pull request merged**